src/shogun/classifier/Perceptron.cpp

```diff
@@ -17,22 +17,14 @@
 using namespace shogun;
 
 CPerceptron::CPerceptron()
-: CLinearMachine()
+: CIterativeLinearMachine()
 {
 	init();
 }
 
-CPerceptron::CPerceptron(CDotFeatures* traindat, CLabels* trainlab)
-: CLinearMachine()
-{
-	init();
-	set_features(traindat);
-	set_labels(trainlab);
-}
-
 void CPerceptron::init()
 {
-	max_iter = 1000;
```

Comment:
keep in mind this will blow the integration test

Reply:
this was temp.. we will remove it

Reply:
i was actually wrong, if the number is the same this is fine